spec/controllers/vm_infra_controller_spec.rb

```diff
@@ -74,8 +74,8 @@
 
   # http://localhost:3000/vm_infra/show/10000000000449
   it 'can display VM details for vm with ems_custom_attributes and a null attribute name' do
-    vm_vmware.ems_custom_attributes.push(custom_attr1, custom_attr2)
-    expect(controller).to receive(:identify_record).and_return(vm_vmware)
+    vm_vmware.ems_custom_attributes = [custom_attr1, custom_attr2]
```

Comment:
`ems_custom_attributes` is an association.
I think `<<` may be a better approach

Reply:
ok, so the issue is the custom attribute was created with a `nil` source.
If you create the attribute with `:source => "VC"`, then this works great.

I do remember warnings that scopes will not properly pass across for `collection.create`. I am not sure if this is related.

Reply:
WAT.  That's crazy.  So, why does it work with assignment?

Reply:
Nice find.  It's interesting that I still need the `identify_record` `twice` change below.